net-libs/rpcsvc-proto/rpcsvc-proto-1.4.4-r1.ebuild

```diff
@@ -27,13 +27,12 @@ src_prepare() {
 	# Bug 718138, 870031, 870061.
 	local x cpp=
 	for x in {${CHOST}-,}{,clang-}cpp; do
-		if type -P "${x}" >/dev/null; then
-			cpp=${x}
+		if cpp="$(type -P "${x}")"; then
```

Comment:
i'd break this into two lines

Reply:
This package has a horrible history (see the referenced bugs) for why it can't use `tc-getCPP`.

Reply:
fwiw `$(tc-getPROG CPP cpp)` typically works fine, it won't give `gcc -E` unless users have excplitly set `CPP="gcc -E"` (and clang profiles uses `CPP="clang-cpp"`)

Reply:
> fwiw `$(tc-getPROG CPP cpp)` typically works fine, it won't give `gcc -E` unless users have excplitly set `CPP="gcc -E"` (and clang profiles uses `CPP="clang-cpp"`)

Please document this as being invalid or get otherwise Ago to stop tinderboxing it.

Reply:
I was working on switching our ChromeOS repo to declaring `CPP=clang-cpp`: https://chromium-review.googlesource.com/c/chromiumos/overlays/chromiumos-overlay/+/5280665/3/profiles/features/llvm/make.defaults#3, but ran into some issues with other ebuilds.

> i'd break this into two lines

I personally always do this type of assignment on one line. When running with `bash -e` (not that portage does this), the naked assignment will abort if it encounters an error vs being handled correctly in the `if` condition. If there is a strong opinion I can change it though.

Would you prefer:
```
cpp="$(type -P "${x}")"
[[ -n "${cpp}" ]] && break
```

or

```
cpp="$(type -P "${x}")"
[[ $? -eq 0 ]] && break
```

?

Reply:
> one line. When running with `bash -e` (not that portage does this), the naked assignment will abort if it encounters an error vs being handled correctly in the if condition.

ah, didn't think of that - fair enough

> If there is a strong opinion I can change it though.

no strong opinion either; and, on top of that, thinking about it the way you presented above clears up the style issue that jumped at me initially.  feel free to leave as is